src/lib/http/proxy-agent.js

```diff
@@ -0,0 +1,85 @@
+// 5. NO_PROXY is set along with VIP_PROXY_OTHER ENABLED and HTTP_PROXY OR HTTPS_PROXY: An HTTP_PROXY/HTTPS_PROXY is returned assuming NO_PROXY is not applicable to the URL
+function createProxyAgent( url ) {
+	const HTTPS_PROXY = process.env.HTTPS_PROXY || process.env.https_proxy || null;
+	const HTTP_PROXY = process.env.HTTP_PROXY || process.env.http_proxy || null;
+	const VIP_PROXY = process.env.VIP_PROXY || process.env.vip_proxy || null;
+	const NO_PROXY = process.env.NO_PROXY || process.env.no_proxy || null;
+
+	// VIP Socks Proxy should take precedence, should be fully backward compatible
+	if ( VIP_PROXY ) {
+		return new SocksProxyAgent( VIP_PROXY );
+	} else if ( process.env.VIP_PROXY_OTHER_ENABLED && ! CoveredInNoProxy( url, NO_PROXY ) && ( HTTPS_PROXY || HTTP_PROXY ) ) {
+		return GetWebProxyAgentBasedOnProtocol( url, HTTPS_PROXY, HTTP_PROXY );
+	}
+	// If no environment variables are set, the no proxy is in effect, or if the proxy enable is not set return null (equivilant of no Proxy agent)
+	return null;
+}
+
+// Determine if a NO_PROXY variable is applicable to a given URL
+// Parameters:
+//	- url (string): absolute desintation URL (including the protocol)
+//	- noProxyString (string | null): string representation of the environments current NO_PROXY or no_proxy variable (NO_PROXY takes precedence)
+// Returns:
+//	- (boolean) true/false depending on result
+// NO_PROXY Rules (based directly on underlying dependency):
+// 1. * (alone): proxy nothing
+// 2. *.site: do not proxy any subdomain of a domain
+// 3. abc.com: do not proxy www.abc.com, abc.com, etc.
+// See getProxyForUrl on npmjs.org for full "ruleset"
+function CoveredInNoProxy( url, noProxyString ) {
+	// If the NO_PROXY env variable is not set, then the URL is not covered in the NO_PROXY (utility below does not handle this case)
+	if ( ! noProxyString ) {
+		return false;
+	}
+	// If getProxyForUrl returns an empty string, then the host should not be proxied
+	// This isn't the most straight forward way to determine if a NO_PROXY is applicable, but the only package I could find that is relatively new and maintained
+	return getProxyForUrl( url ) === '';
+}
+
+// Returns either an HTTPS or HTTP ProxyAgent based on the protocol of the given URL
+// Parameters:
+//	- url (string): absolute desintation URL (including the protocol)
+//	- httpsProxy (string | null): string location of https proxy (or null)
+//	- httpProxy (string | null): string location of http proxy (or null)
+// Requires:
+//	- a no_proxy variable is not applicable to the url
+//	- either the environment variable HTTP_PROXY or HTTPS_PROXY is set
+// Returns:
+//	- Either an instance of httpsProxyAgent or httpProxy agent depending on passed in values
+function GetWebProxyAgentBasedOnProtocol( url, httpsProxy, httpProxy ) {
+	const protocol = url.substr( 0, 5 );
+	if ( protocol !== 'https' && protocol !== 'http:' ) {
```

Comment:
I think I misunderstood on some level the differences between HTTPS and HTTP proxies. For now, I removed all references to using an HTTP proxy (which simplified the implementation). If you think using HTTP for development is going to happen, we can figure out how to enable in development only (or something along those lines).

But as you said, since production will always be HTTPS, there is little need for HTTP proxy support.